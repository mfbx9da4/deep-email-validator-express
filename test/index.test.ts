import Axios from 'axios'
import _ from 'lodash'
import server from '../src/index'
const axios = Axios.create({
  baseURL: 'http://localhost:8080',
})

describe('name', () => {
  afterAll(() => {
    server.close()
  })
  it('test endpoint', async () => {
    const res = await axios('/email/validate?email=davidadler@gmail.com')
    expect(res.data.valid).toBe(true)
    expect(res.data).toMatchSnapshot()
  })

  it('test endpoint POST', async () => {
    const res = await axios.post('/email/validate', {
      email: 'davidadler@gmail.com',
    })
    expect(res.data.valid).toBe(true)
    expect(res.data).toMatchSnapshot()
  })
})
