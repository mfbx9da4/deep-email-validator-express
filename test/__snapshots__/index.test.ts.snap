// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`name test endpoint 1`] = `
Object {
  "valid": true,
  "validators": Object {
    "disposable": Object {
      "valid": true,
    },
    "mx": Object {
      "valid": true,
    },
    "regex": Object {
      "valid": true,
    },
    "smtp": Object {
      "valid": true,
    },
    "typo": Object {
      "valid": true,
    },
  },
}
`;

exports[`name test endpoint POST 1`] = `
Object {
  "valid": true,
  "validators": Object {
    "disposable": Object {
      "valid": true,
    },
    "mx": Object {
      "valid": true,
    },
    "regex": Object {
      "valid": true,
    },
    "smtp": Object {
      "valid": true,
    },
    "typo": Object {
      "valid": true,
    },
  },
}
`;
